doc/exception_interrupts.rst

```diff
@@ -94,13 +99,20 @@ Ibex can trigger an exception due to the following exception causes:
 The illegal instruction exception, instruction access fault, LSU error exceptions and ECALL instruction exceptions cannot be disabled and are always active.
 
 
-Handling
---------
+Nested Interrupt/Exception Handling
+-----------------------------------
 
-Ibex does support nested interrupt/exception handling.
-Exceptions inside interrupt/exception handlers cause another exception.
-However, exceptions during the critical part of your exception handlers, i.e. before having saved the ``mepc`` and ``mstatus``, will cause those CSRs to be overwritten.
-Interrupts during interrupt/exception handlers are thus disabled by default, but can be explicitly enabled if desired.
+Ibex does support nested interrupt/exception handling in software.
+However, interrupts/exceptions during the critical part of the handler, i.e. before software has saved the ``mepc`` and ``mstatus``, will cause those CSRs to be overwritten.
+The hardware thus automatically disables interrupts upon entering an interrupt/exception handler.
+If desired, software can explicitly enable interrupts by setting ``mstatus``.MIE to 1'b1 from within the handler.
+However, software should only do this after saving ``mepc`` and ``mstatus``.
+There is no limit on the maximum number of nested interrupts.
+Note that, after enabling interrupts by setting ``mstatus``.MIE to 1'b1, the current handler will be interrupted also by lower priority interrupts.
```

Comment:
Would it be possible to include a couple lines of pseudo-code how to handle nested interrupts in software?

```c
isr_handle_nested_interrupts() {
  // Save mpec and mstatus to stack
  mepc_bak = mepc;
  mstatus_bak = mstatus;

  // Re-enable interrupts (optional)
  mstatus.MIE = 1;

  // Handle interrupt
  // This code block can be interrupted by other interrupts, including lower-priority ones.

  // XXX: Include example code how to get only higher priority interrupts.

  // XXX: Do you need to disable interrupts before restoring?

  // Restore mepc and mstatus
  mpec = mepc_bak;
  mstatus = mstatus_bak;
}

Reply:
Excellent idea! I adapted your code snipped a bit and added that to the documentation.